frontend/src/pages/journal/journal.tsx

```diff
@@ -24,18 +24,25 @@ const Journal: React.FC = () => {
   const { isSidebarShown, setIsSidebarShow } = useSidebarState();
   const { filter, setFilter } = useSearch();
 
-  const { selectedJournalEntry } = useAppSelector(({ journal }) => {
-    return {
-      selectedJournalEntry: journal.selectedJournalEntry,
-    };
-  });
+  const { selectedJournalEntry, journalEntriesDataStatus } = useAppSelector(
+    ({ journal }) => {
+      return {
+        selectedJournalEntry: journal.selectedJournalEntry,
+        journalEntriesDataStatus: journal.journalEntriesDataStatus,
+      };
+    },
+  );
 
-  const hasSelectedNote = Boolean(id) && selectedJournalEntry;
+  const isLoading =
+    journalEntriesDataStatus === DataStatus.IDLE ||
+    journalEntriesDataStatus === DataStatus.PENDING ||
+    selectedJournalEntry?.id.toString() !== id;
```

Comment:
Not sure that is a suitable condition for isLoading

Reply:
I removed it, it is really redundant at that point